### PR TITLE
ML word2vect api

### PR DESCRIPTION
**Pull request by mkarmona**

### CODE REVIEW

**Review by cmalangone — DISMISSED** [review]

File: app/models/ElasticRetriever.scala [hunk 1]
```diff
@@ -183,6 +183,68 @@ class ElasticRetriever @Inject()(client: ElasticClient,
...
+          q.sortBy(sortByFields:_*)
+
+      }
+
+      logger.debug(client.show(qq))
+      qq
+    }
+
+    elems.map {
+      case _: RequestFailure                       => (IndexedSeq.empty, 0, None)
```
Comment (JarrodBaker) [inline app/models/ElasticRetriever.scala:218]: Failing silently: we should log something here so we can figure out what's happening.

File: app/models/ElasticRetriever.scala [hunk 2]
```diff
@@ -183,6 +183,68 @@ class ElasticRetriever @Inject()(client: ElasticClient,
...
+        val totalHits = results.result.totalHits
+
+        val mappedHits = hits
+          .map(jObj => {
+            buildF(jObj)
+          })
+          .withFilter(_.isDefined)
+          .map(_.get)
+
+        val hasNext = !(hits.size < pageSize) && pageSize > 0
```
Comment (JarrodBaker) [inline app/models/ElasticRetriever.scala:235]: I have a feeling that I might have written this line, but looking at it now it really should be 
``` 
val hasNext = pageSize < hits.size && pageSize > 0
```

**Review by JarrodBaker — CHANGES_REQUESTED** [review]
We just need to log why the ES request filed, otherwise looks good. 

The gql/Objects file is getting pretty big, and we're accumulating duplicated code. Maybe next time we thing about refactoring we can consider as a team how we can clean it up. Eg. Arguments is already broken out into a separate Object, maybe Fields can be too.

File: app/models/ElasticRetriever.scala (hunk 1 above)
Comment (cmalangone) [reply app/models/ElasticRetriever.scala:218]: Good catch

**Comment by mkarmona** [comment]
@JarrodBaker @cmalangone the gql needs a proper refactor and putting in practice more functional programming and better abstractions. Also, the elasticretriever can be better refactored as I had putting multiple functions with slightly different functioning but those can be fairly refactored into something more robust and function passing way style

**Review by cmalangone — APPROVED** [review]

**Pull request merged**